signer/src/transaction_signer.rs

```diff
@@ -1024,7 +1008,12 @@ where
             .get(id)
             .ok_or(Error::MissingStateMachine)?;
 
-        let encrypted_dkg_shares = state_machine.get_encrypted_dkg_shares(&mut self.rng)?;
+        let StateMachineId::Dkg(started_at) = id else {
+            return Err(Error::TypeConversion);
```

Comment:
nit: maybe using `UnexpectedStateMachineId` is better suited here

Reply:
Oh I just forgot to get back to this one and glossed over it all when opening the PR. Will update.

Reply:
Fixed in https://github.com/stacks-network/sbtc/pull/1316/commits/36424ecb8063aeff3df2c5f0558bc3bba8b18ae7.